modules/p2/ding_serendipity/ding_serendipity.module

```diff
@@ -350,6 +352,25 @@ function ding_serendipity_fetch(array $context = NULL) {
 }
 
 /**
+ * Flush serendipity cache of one user.
+ *
+ * @param int $uid
+ *   The user uid.
+ */
+function ding_serendipity_user_cache_clear($uid) {
+  cache_clear_all("ding_serendipity_" . $uid . "_", "cache", TRUE);
+}
+
+/**
+ * Flush serendipity cache of current user if list elements change.
+ */
+function ding_serendipity_entity_presave($entity, $type) {
+  if (isset($entity->type) && $entity->type == "ding_list_element" && user_is_logged_in() && isset($entity->uid)) {
+    ding_serendipity_user_cache_clear($entity->uid);
```

Comment:
This clear the cache for the owner of the element - not the current user as suggested by the DocBlock.

That may be intentional but if that is the case why is there a need to check if the current user is logged in?

I would expect one of the following:

Clear serendipity cache for owner of list elements on when changed by whoever.

```php
if (isset($entity->type) && $entity->type == "ding_list_element" && isset($entity->uid)) {
```

or

Clear serendipity chance for current user when changing list elements:

```php
global $user;
if (isset($entity->type) && $entity->type == "ding_list_element" && !empty($user->uid)) {
```